internal/services/runservice/api/api.go

```diff
@@ -311,6 +311,108 @@ func sendLogs(w http.ResponseWriter, r io.Reader) error {
 	}
 }
 
+type LogsDeleteHandler struct {
+	log *zap.SugaredLogger
+	e   *etcd.Store
+	ost *objectstorage.ObjStorage
+	dm  *datamanager.DataManager
+}
+
+func NewLogsDeleteHandler(logger *zap.Logger, e *etcd.Store, ost *objectstorage.ObjStorage, dm *datamanager.DataManager) *LogsDeleteHandler {
+	return &LogsDeleteHandler{
+		log: logger.Sugar(),
+		e:   e,
+		ost: ost,
+		dm:  dm,
+	}
+}
+
+func (h *LogsDeleteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	q := r.URL.Query()
+
+	runID := q.Get("runid")
+	if runID == "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+	taskID := q.Get("taskid")
+	if taskID == "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	_, setup := q["setup"]
+	stepStr := q.Get("step")
+	if !setup && stepStr == "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+	if setup && stepStr != "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	var step int
+	if stepStr != "" {
+		var err error
+		step, err = strconv.Atoi(stepStr)
+		if err != nil {
+			http.Error(w, "", http.StatusBadRequest)
+			return
+		}
+	}
+
+	if err, deleteError := h.deleteTaskLogs(ctx, runID, taskID, setup, step, w); err != nil {
+		h.log.Errorf("err: %+v", err)
+		if deleteError {
+			switch err.(type) {
```

Comment:
just use `httpError()` function. So it'll handle the error kind and convert it to the right http error.

I know you just copied other functions but they will need to be updated (another PR) to use httpError.